pygments/lexers/asm.py

```diff
@@ -55,7 +55,9 @@ class GasLexer(RegexLexer):
             (number, Number.Integer),
             (register, Name.Variable),
             (r'[\r\n]+', Text, '#pop'),
-            (r'[;#].*?\n', Comment, '#pop'),
+            (r'([;#]|//).*?\n', Comment, '#pop'),
+            (r'/[*].*?[*]/', Comment),
+            (r'/[*].*?\n[\w\W]*?[*]/', Comment, '#pop'),
```

Comment:
Is this `#pop` needed here? I'm looking at the C lexer, and it uses
```python
(r'/[*](.|\n)*?[*]/', Comment.Multiline),
```

for multi-line comments, which I presume would be sufficient  for GAS as well? Plus there are two separate rules for what should be the same type of comment?

Reply:
Yes, I think it is necessary to `#pop` here.

It is because GAS thinks the current instruction is ended and it will parse the following code as the next instruction when GAS sees `\n` in multi-line comments. Please see the first two examples I mentioned in the pull request for detail.

The case is different in C language. It will never care whether there is `\n` in multi-line comments or not.